x/nft/base/handler.go

```diff
@@ -0,0 +1,161 @@
+package base
+
+import (
+	"fmt"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+	"github.com/iov-one/weave/x/nft"
+	"github.com/iov-one/weave/x/nft/blockchain"
```

Comment:
This looks like it could easily cause a circular import if not careful.
In this codebase, I think I have enforced that packages only import from their parents, so there is a one-way flow.
Except for some packages like `error` that are meant to be imported by everyone.

I was quite inspired by this article:
https://medium.com/@benbjohnson/standard-package-layout-7cdbc8391fc1

Since the subpackages are only needed for the RegisterRoutes issue, and I found that a bit inflexible,
I would request to invert the logic somehow so the subpackages import the base `nft` package and then register their buckets on the bucketMap

Maybe add a RegisterBucket() method that is called in init of all the subpackages? Or something cleaner?

Reply:
@ethanfrey So that is why I created a subpackage. This is kind of a "shared" package between all of them because of shared buckets and members, but it is created specifically to avoid circular dependency hell, otherwise I would have introduced it within parent "nft" dir.

Let me know if that makes sense.

We don't need to protect "base" package from circular deps, because it's actually designed to avoid those!

Reply:
I should rename it to "shared" maybe?

Reply:
So the idea behind "base" package is that it would never have it's own bucket and would basically have shared handlers.

I will take a look today to see how we can avoid doing the bucket stuff here and have some common structure hold them.

Reply:
I am fine with the base package. I think I missed that part.

However, I would say the concrete implementations should import base, rather than base importing concrete implementations.

If this change is too much work for now, let's push that to another PR, like with the protobuf stuff above